pandas/core/arrays/base.py

```diff
@@ -335,7 +335,7 @@ def __iter__(self):
         for i in range(len(self)):
             yield self[i]
 
-    def __eq__(self, other: Any) -> ArrayLike:
+    def __eq__(self, other: Any) -> ArrayLike:  # type: ignore[override] # NOQA
```

Comment:
im not familiar with the `ignore[overrride] # NOQA` pattern.  do we use that elsewhere?

Reply:
I thought I posted the mypy issue, but apparently forgot.

So the problem here is that mypy expects a "bool" return for `__eq__` since the base `object` is typed that way in the stubs. In https://github.com/python/mypy/issues/2783, the recommended way to solve this is the above with `# type: ignore[override]`. 

But flake8 doesn't like that, hence also the #NOQA (https://github.com/PyCQA/pyflakes/issues/475). 

This was already done in another PR as well:

https://github.com/pandas-dev/pandas/blob/7aa710a4d4d3b8515a723a366e50f49b327adc23/pandas/core/frame.py#L5062